src/Npgsql/NpgsqlConnection.cs

```diff
@@ -723,8 +720,7 @@ internal bool IsSecure
             get
             {
                 CheckConnectionOpen();
-                Debug.Assert(Connector != null);
-                return Connector.IsSecure;
+                return Connector!.IsSecure;
```

Comment:
It's OK, but the better way to do this is to have CheckConnectionOpen return the connector - this way no bang needed (and it even becomes an expression-bodied property). But we can really defer for later.